audio/src/lib.rs

```diff
@@ -29,8 +30,65 @@ pub use fetch::{
     READ_AHEAD_BEFORE_PLAYBACK_ROUNDTRIPS, READ_AHEAD_BEFORE_PLAYBACK_SECONDS,
     READ_AHEAD_DURING_PLAYBACK_ROUNDTRIPS, READ_AHEAD_DURING_PLAYBACK_SECONDS,
 };
+use std::fmt;
+
+pub enum AudioPacket {
+    Samples(Vec<i16>),
+    Bytes(Vec<u8>),
+}
+
+impl AudioPacket {
+    pub fn samples(&self) -> &[i16] {
+        match &*self {
```

Comment:
If you go with pattern matching instead of using this functions (as @plietar suggested below), you could also remove this functions entirely. At least I wouldn't expect such harmless-looking functions to panic.

Reply:
@Johannesd3 - understood but back to my comment, then I would need to modify all backends to do the matching and I was trying to avoid that like plague

Reply:
@Johannesd3 I actually think these functions are useful inside the backends that only support one kind of packet. Of course each backend could also match and do the panic themselves, but it's cleaner in a version.

Reply:
Done